presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8447,6 +8447,14 @@ public void testSetAgg()
                 "select count(distinct orderdate) from orders");
     }
 
+    @Test
+    public void testRedundantLambda()
+    {
+        assertQuery(
+                "SELECT x, reduce(x, 0, (s, x) -> s + x, s -> s), reduce(x, 0, (s, x) -> s + x, s -> s) FROM (VALUES (array[1, 2, 3])) t(x)",
+                "SELECT array[1, 2, 3], 6, 6");
+    }
```

Comment:
I tried this query out on the master with the CSE enabled and it is succeeding. Is there a case which fails on the master but succeeds with this fix?

Reply:
My mistake, the issue on master but is not in a release.